### PR TITLE
Fix lots of broken links and line numbers in documentation

### PR DESCRIPTION
**Pull request by white238**

### CODE REVIEW

File: docs/external_dependencies.rst [hunk 1]
```diff
@@ -51,15 +51,23 @@ To accomplish this BLT provides a ``DEPENDS_ON`` option for the
 The ``blt_register_library`` macro allows you to expose any external dependency,
 including any additional includes, libraries, compile flags, link flags, defines, etc that are required to use the dependency via a single name.
 
-For example, instead adding the include directories and libraries necessary to use a dependency *zzz* directly to your CMake executable target, if  *zzz* was
-registered with ``blt_register_library``, you can simply use 
+For example, instead adding the include directories, libraries, and other information
```
Comment (kennyweiss) [inline docs/external_dependencies.rst]: instead \*of\* ?

File: docs/external_dependencies.rst [hunk 2]
```diff
@@ -51,15 +51,23 @@ To accomplish this BLT provides a ``DEPENDS_ON`` option for the
...
+necessary to use an external dependency *axom* directly to your CMake target,
```
Comment (kennyweiss) [inline docs/external_dependencies.rst]: This sentence is a bit awkward.  (it was awkward in the original as well).

File: docs/external_dependencies.rst [hunk 3]
```diff
@@ -71,16 +79,16 @@ BLT also supports using ``blt_register_library`` to provide additional options f
...
    :class: hint
 
-   A macro to register external libraries and dependencies with blt.
-   The named target can be added to the ``DEPENDS_ON`` argument of other blt macros, like ``blt_add_library`` and ``blt_add_executable``.  
+   A macro to register external libraries and dependencies with BLT.
+   The named target can be added to the ``DEPENDS_ON`` argument of other BLT macros, like ``blt_add_library`` and ``blt_add_executable``.  
 
 
-You have already seen one use of ``DEPENDS_ON`` of a blt
+You have already seen one use of ``DEPENDS_ON`` of a BLT
```
Comment (kennyweiss) [inline docs/external_dependencies.rst]: of a BLT -> \*for\* a BLT ?

File: docs/external_dependencies.rst (hunk 2 above)
Comment (white238) [reply docs/external_dependencies.rst]: should be better now.

**Pull request merged**